tests/10apidoc/01register.pl

```diff
@@ -126,7 +128,75 @@ sub matrix_register_user
    });
 }
 
-push @EXPORT, qw( local_user_fixture local_user_fixtures );
+push @EXPORT, qw( matrix_register_user_via_secret );
+
+sub matrix_register_user_via_secret
+{
+   my ( $http, $uid, %opts ) = @_;
+
+   my $password = $opts{password} // "an0th3r s3kr1t";
+   my $admin = $opts{admin} // 0;
+
+   defined $uid or
+      croak "Require UID for matrix_register_user_via_secret";
+
+   my $hmac = Digest::HMAC_SHA1->new("reg_secret");
+   $hmac->add( $uid );
+   $hmac->add( "\0" );
+   $hmac->add( $password );
+   $hmac->add( "\0" );
+   $hmac->add( $admin ? "admin" : "notadmin" );
+
+   my $mac = $hmac->hexdigest;
```

Comment:
Actually, looking at the module there's a more convenient "all-in-one" function of `hmac_sha1_hex()` you could use instead.

    use Digest::HMAC_SHA1 qw( hmac_sha1_hex );

    my $mac = hmac_sha1_hex(
       join( "\0", $uid, $password, $admin ? "admin" : "notadmin" ),
       "reg_secret"
    );

That might look shorter.